Content.Server/Singularity/EntitySystems/ContainmentFieldGeneratorSystem.cs

```diff
@@ -10,6 +14,45 @@ public override void Initialize()
             base.Initialize();
 
             SubscribeLocalEvent<ContainmentFieldGeneratorComponent, PhysicsBodyTypeChangedEvent>(BodyTypeChanged);
+            SubscribeLocalEvent<ContainmentFieldComponent, StartCollideEvent>(HandleFieldCollide);
+            SubscribeLocalEvent<ContainmentFieldGeneratorComponent, StartCollideEvent>(HandleGeneratorCollide);
+            SubscribeLocalEvent<ParticleProjectileComponent, StartCollideEvent>(HandleParticleCollide);
+        }
+
+        private void HandleParticleCollide(EntityUid uid, ParticleProjectileComponent component, StartCollideEvent args)
+        {
+            if (args.OtherFixture.Body.Owner.TryGetComponent<SingularityGeneratorComponent>(out var singularityGeneratorComponent))
+            {
+                singularityGeneratorComponent.Power += component.State switch
+                {
+                    ParticleAcceleratorPowerState.Standby => 0,
+                    ParticleAcceleratorPowerState.Level0 => 1,
+                    ParticleAcceleratorPowerState.Level1 => 2,
+                    ParticleAcceleratorPowerState.Level2 => 4,
+                    ParticleAcceleratorPowerState.Level3 => 8,
+                    _ => 0
+                };
+
+                component.Owner.Delete();
```

Comment:
You know you can just use the `EntityManager` for this, ye? Also turn it into a `QueueDelete`, please